types/index.d.ts

```diff
@@ -322,6 +331,8 @@ export interface ChannelListProps {
     e: Client.Event,
   ): void;
   setActiveChannelOnMount?: boolean;
+  /** Optional channel list filter */
+  filterFunction?: (channels: Array<Client.Channel>) => Array<Client.Channel>;
```

Comment:
you can make these StreamChatChannel now that we have that type with the generics